great_expectations/checkpoint/checkpoint.py

```diff
@@ -59,7 +59,7 @@ def __init__(
         class_name: Optional[str] = None,
         run_name_template: Optional[str] = None,
         expectation_suite_name: Optional[str] = None,
-        batch_request: Optional[Union[BatchRequest, dict]] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
```

Comment:
@NathanFarmer Is this needed?  Since `RuntimeBatchRequest` extends `BatchRequest`, we could simply use `cast(RuntimeBatchRequest, batch_request)` to avoid type hints warnings in PyCharm.  Or is this having to do with something deeper, such as the presence of a `DataFrame` in `batch_data`?

Reply:
@alexsherstinsky ✅ I removed `RuntimeBatchRequest` because I am not handling `batch_request` passed directly to `Checkpoint` in this PR (I considered it at one point). I'm not familiar with how exactly `cast(RuntimeBatchRequest, batch_request)` would be used though. This is referring to the case where a user passes `batch_request=RuntimeBatchRequest(...)` into the top level, instead of through `validations` at `Checkpoint` initialization.